controllers/iamrole_controller.go

```diff
@@ -415,7 +417,7 @@ func (r *IamroleReconciler) UpdateStatus(ctx context.Context, iamRole *iammanage
 
 	//if wait time is specified, requeue it after provided time
 	if len(requeueTime) == 0 {
-		requeueTime[0] = 0
+		requeueTime[0] = defaultRequeueTime
```

Comment:
Won't this default back to 30s when re-queue time is `0`?

Reply:
This function signature is `func (r *IamroleReconciler) UpdateStatus(ctx context.Context, iamRole *iammanagerv1alpha1.Iamrole, status iammanagerv1alpha1.IamroleStatus, requeueTime ...float64) (ctrl.Result, error)`

so `requeueTime` is array, and if `requeueTime` length is 0, it will be assigned first value to be defaultRequeueTime.

Would you recommend to put requeueTime as required field?

Reply:
Yea, I think it should be required, can just pass in defaults.

Reply:
Thank you @kevdowney for feedback, I have updated function signature and update all callers.